src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -68,7 +69,7 @@ def __ne__(self, other):
 def extract_mds_ranks_from_spec(mds_rank_spec):
     if not mds_rank_spec:
         return MDS_RANK_ALL
-    match = re.match(r'^(\d[,\d]*)$', mds_rank_spec)
+    match = re.match(r'^\d+(,\d+)*$', mds_rank_spec)
```

Comment:
should the regex be:

```
match = re.match(r'^(\d+[,\d+]*)$', mds_rank_spec)
```

Reply:
I tried above suggested regex but still facing same issue.

nshelke@fedora build]$ ./bin/ceph fs perf stats --mds_rank=1,
2021-12-08T16:33:29.368+0530 7faebe77b640 -1 WARNING: all dangerous and experimental features are enabled.
2021-12-08T16:33:29.375+0530 7faebe77b640 -1 WARNING: all dangerous and experimental features are enabled.
Error EINVAL: invalid literal for int() with base 10: ''

Reply:
That is *not* due to the regex, the trailing "," ends up as an empty string in the list comprehension below.

Reply:
.. since the "," is optional in the regex.

Reply:
or you can modify the regex to not capture the last "," if there is any...

Reply:
Discussed offline.
Marking discussion resolved.

Please find few test outputs below
1. mds_rank=10,
$ ./bin/ceph fs perf stats --mds_rank=10,
2021-12-08T19:20:36.631+0530 7f77e98fb640 -1 WARNING: all dangerous and experimental features are enabled.
2021-12-08T19:20:36.639+0530 7f77e98fb640 -1 WARNING: all dangerous and experimental features are enabled.
Error EINVAL: invalid mds filter spec: 10,

2. mds_rank=10,a
$ ./bin/ceph fs perf stats --mds_rank=10,a
2021-12-08T19:27:58.686+0530 7f9c61251640 -1 WARNING: all dangerous and experimental features are enabled.
2021-12-08T19:27:58.693+0530 7f9c61251640 -1 WARNING: all dangerous and experimental features are enabled.
Error EINVAL: invalid mds filter spec: 10,a